conv/convert.go

```diff
@@ -51,53 +52,77 @@ func IsFloat64AJSONInteger(f float64) bool {
 	return diff/math.Min(fa+ga, math.MaxFloat64) < epsilon
 }
 
-var evaluatesAsTrue map[string]struct{}
-
-func init() {
-	evaluatesAsTrue = map[string]struct{}{
-		"true":     {},
-		"1":        {},
-		"yes":      {},
-		"ok":       {},
-		"y":        {},
-		"on":       {},
-		"selected": {},
-		"checked":  {},
-		"t":        {},
-		"enabled":  {},
+// ConvertFloat turns a string into a float numerical value.
+func ConvertFloat[T Float](str string) (T, error) {
+	// NOTE: [unsafe.SizeOf] simply returns the size in bytes of the value.
+	// For primitive types T, the generic stencil is precompiled and this value
+	// is resolved at compile time, resulting in an immediate call to [strconv.ParseFloat].
+	var v T
+	f, err := strconv.ParseFloat(str, int(unsafe.Sizeof(v))*8)
```

Comment:
This is good, but you are using unsafe.Sizeof in various place here in this file

Please consider:

- creating a sizeof.go file
- add a sizeOfNumber func
- add this exact comment
- call sizeOfNumber everywhere else

This way, only sizeof.go will import the unsafe package and it would be clearer

Reply:
Did you see this before merging ?

@fredbi

Reply:
ok thanks. I'v been considering that wrapper func too. Perhaps in a follow-up. Now it is time to merge and move on to the next series of packages

Reply:
I opened

- https://github.com/go-openapi/swag/issues/110